modules/siddhi-extensions/imageprocessing-objectdetection/src/main/java/org/wso2/siddhi/extension/objectdetection/ObjectDetectionExtension.java

```diff
@@ -0,0 +1,173 @@
+
+/**
+ * The following class represents a functional siddhi extension. The extension allows to detect
+ * objects of a given image using OpenCV functions. The object detection is done using a method
+ * called Haar Classifier in image processing context. An xml file is used to describe the type of
+ * object needs to be detected. This xml file is also known as a cascade file.
+ * </p>
+ * The extension take 2 input arguments.
+ * 1. The image as a hex string.
+ * 2. The cascade file path for the object detection.
+ * </p>
+ * Suppressing "UnusedDeclaration" as this is referenced through CEP.
+ */
+@SuppressWarnings("UnusedDeclaration")
+@SiddhiExtension(namespace = "imageprocessorobjectdetection", function = "count")
+public class ObjectDetectionExtension extends FunctionExecutor {
+
+    /**
+     * The logger to log information, warnings or errors.
+     */
+    private static final Logger log = Logger.getLogger(ObjectDetectionExtension.class);
+
+    /**
+     * The return type for the functional extension.
+     */
+    private Attribute.Type returnType;
+
+    /**
+     * Loads the native libraries for OpenCV.
+     * @see <a href="https://github.com/PatternConsulting/opencv#api">OpenCV API loading.</a>
+     */
+    static {
+        OpenCV.loadLibrary();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void destroy() {
+
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Type getReturnType() {
+        return returnType;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void init(Type[] types, SiddhiContext siddhiContext) {
+        returnType = Attribute.Type.LONG;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected Object process(Object inputData) {
+        long detectedObjectCount;
+        if (inputData instanceof Object[]) {
+            Object[] arguments = (Object[]) inputData;
+            if (arguments.length == 2) {
+                if (arguments[0] instanceof String && arguments[1] instanceof String) {
+                    String imageHex = (String) arguments[0];
+                    String cascadeFilePath = (String) arguments[1];
+                    detectedObjectCount = this.detectObjects(imageHex, cascadeFilePath);
+                } else {
+                    throw new IllegalArgumentException(
+                            "2 String arguments of the hex string of the image and the cascade " +
+                            "path is expected.");
+                }
+            } else {
+                throw new IllegalArgumentException(
+                        "2 String arguments of the hex string of the image and the cascade path " +
+                        "is expected.");
+            }
+        } else {
+            throw new IllegalArgumentException(
+                    "2 String arguments of the hex string of the image and the cascade path is " +
+                    "expected.");
+        }
```

Comment:
Added more context for the exceptions. See fix in https://github.com/hemikak/siddhi/commit/bb4059c56810d708c9e91cd7283b5891a5af8c60